flink-java/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -310,6 +310,9 @@ public int getFieldIndex(String fieldName) {
 
 	@Override
 	public TypeSerializer<T> createSerializer(ExecutionConfig config) {
+		if(config.isForceKryoEnabled()) {
+			return new GenericTypeInfo<T>(this.typeClass).createSerializer(config);
```

Comment:
Yes, that would save the creation of the GenericTypeInfo.
I did it like this because the createSerializer() method of the GenericTypeInfo is again configurable, creating a Kryo or Avro serializer, depending on the execution config.
But I can also pull the Kryo/Avro switch to the PojoTypeInfo if you think that's nicer.